src/module.rs

```diff
@@ -201,17 +201,19 @@ where
 {
     let mut used = 0usize;
     let mut current: Vec<AnsiString> = Vec::new();
-    let mut chunks: Vec<(Vec<AnsiString>, &FillSegment)> = Vec::new();
+    let mut chunks: Vec<(Vec<AnsiString>, &FillSegment, Option<::nu_ansi_term::Style>)> =
+        Vec::new();
 
     for segment in segments {
+        let prev_style = current.last().map(|s| *s.style_ref());
         match segment {
             Segment::Fill(fs) => {
-                chunks.push((current, fs));
+                chunks.push((current, fs, prev_style));
```

Comment:
Since `chunks` does not impact `current` the `Fill` style will be unavailable to the next item.